### PR TITLE
Update Random Priors Demo

### PR DESCRIPTION
**Pull request by petergchang**
## Description

Continued from [PR61](https://github.com/probml/probml-notebooks/pull/61).

1. Set the adam optimizer learning rate as a parameter for training functions.
2. Vectorized `get_ensemble_predictions()` and the code that generates predictions over a range of beta values using `jax.vmap()`.
3. Fixed an issue in the implementation of `create_train_state()` method that was surreptitiously leaking a Trace of `beta`.
4. Modified the final plot comparing with/without bootstrap and/or prior to `fill_between` 1 and 2 standard deviations, with a larger `beta=30` value. This resulted in a much more clear distinction among the four types of models that closely matches the description given by the [original Kaggle notebook](https://www.kaggle.com/code/gdmarmerola/introduction-to-randomized-prior-functions/notebook):
> Generally, turning bootstrapping off will reduce uncertainty the most (upper right corner), as opposed to turning priors off (lower left corner), but it can vary a bit across different seeds. With both bootstrapping and priors off, there's still a little disagreement between ensemble members due to random initialization of weights, but is a lot less compared to the other models.

### Colab link

https://colab.research.google.com/github/probml/probml-notebooks/blob/main/notebooks/randomized_priors.ipynb

### Issue 
[#708: translate ensembles with random priors demo from keras to JAX](https://github.com/probml/pyprobml/issues/708) 

### Figures

[Gist for Updated Bootstrap/Prior-Dependence Figure](https://gist.github.com/peterchang0414/b894443d24bae64b72a8e4a572fb8604)

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

### CODE REVIEW

**Comment by murphyk** [comment]
Hi @peterchang0414 , I left some comments in https://app.reviewnb.com/probml/probml-notebooks/pull/69/, not sure if you can see them? (I'm new to ReviewNB)

**Comment by petergchang** [comment]
Hello @murphyk, thank you very much for your thoughtful comments!

Following your advice, I made the following revisions for the most recent commit(s):

1. Updated the "Open in Colab" button to point to the probml repo (though I had to use a contrived method to link to the remote repo since I don't have write access).
2. Deleted the unnecessary `beta` parameter from `create_train_state()` function.
3. Modified the `build_ensemble()` such that it first stacks each of the (possibly bootstrapped) training set for each model in ensemble, then `vmap`s the `train()` function to the resulting stack. 
4. The modification led to noticeable speedup, e.g., `%%timeit` for `get_ensemble_predictions()` (2000 epochs) improved from 13s per loop (best of 5) to 5.08s per loop (best of 5) 

For creating a vectorized function for computing predictions for an array of different beta values, I studied the `xmap` tutorial you linked and perused the source code to see how it may help, but failed to see how it would help in this case (the cluttering was even worse in my `xmap` experiments). I'm new to the JAX and I may not yet be grokking `xmap` enough to understand your advice. If you have more hints as to how `xmap` would help here, I'd greatly appreciate it!

Thanks again.

**Comment by murphyk** [comment]
I am not very familar with xmap either, it just seemed like an easy way to name axes, so you know what is being vmap'd over. But it's just syntactic sugar, and is not needed.
